charts/tidb-cluster/templates/monitor.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.monitor.create }}
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  # for backward compatibility, name should not be {{ template "cluster.name" . }}-monitor
+  # or the names of the pods/svcs would be changed, like tidb-cluster-grafana -> tidb-cluster-monitor-grafana
+  name: {{ template "cluster.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ template "chart.name" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: monitor
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  clusters:
+    - name: {{ template "cluster.name" . }}
+      namespace: {{ .Release.Namespace }}
+  persistent: {{ .Values.monitor.persistent }}
+  storage: {{ .Values.monitor.storage }}
+  prometheus:
+    logLevel: {{ .Values.monitor.prometheus.logLevel }}
+    service:
+      type: {{ .Values.monitor.prometheus.service.type }}
+      annotations: {}
```

Comment:
The service annotations should be exposed to users if they have their own configuraition.